src/txrebroadcast.cpp

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2021 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <chainparams.h>
+#include <consensus/consensus.h>
+#include <logging.h>
+#include <miner.h>
+#include <script/script.h>
+#include <txrebroadcast.h>
+#include <validation.h>
+
+/** We rebroadcast 3/4 of max block weight to reduce noise due to circumstances
+ *  such as miners mining priority transactions. */
+static constexpr unsigned int MAX_REBROADCAST_WEIGHT = 3 * MAX_BLOCK_WEIGHT / 4;
+
+/** Default minimum age for a transaction to be rebroadcast */
+static constexpr std::chrono::minutes REBROADCAST_MIN_TX_AGE = 30min;
+
+/** Maximum number of times we will rebroadcast a tranasaction */
```

Comment:
Perhaps add some of the description from the _[rebroadcast] Track rebroadcast attempts_ commit log here. It's not at all obvious why we'd stop rebroadcasting after some time, and that commit log has a good explanation.

Reply:
done, how's it look? [`3f4b664` (#21061)](https://github.com/bitcoin/bitcoin/pull/21061/commits/3f4b664fbc021b1c1ccabbf0f448a208c1463496)